compiler/rustc_codegen_llvm/src/gotoc/hooks.rs

```diff
@@ -74,6 +74,46 @@ fn name_is(tcx: TyCtxt<'tcx>, instance: Instance<'tcx>, expected: &str) -> bool
     false
 }
 
+struct ExpectFail;
+impl<'tcx> GotocHook<'tcx> for ExpectFail {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        let def_path = tcx.def_path(instance.def.def_id());
+        if let Some(data) = def_path.data.last() {
+            match data.data.name() {
+                DefPathDataName::Named(name) => {
+                    return name.to_string().starts_with("__VERIFIER_expect_fail");
+                }
+                DefPathDataName::Anon { .. } => (),
+            }
+        }
+        false
+    }
+
+    fn handle(
+        &self,
+        tcx: &mut GotocCtx<'tcx>,
+        _instance: Instance<'tcx>,
+        mut fargs: Vec<Expr>,
+        _assign_to: Option<Place<'tcx>>,
+        target: Option<BasicBlock>,
+        span: Option<Span>,
+    ) -> Stmt {
+        assert_eq!(fargs.len(), 2);
+        let target = target.unwrap();
+        let cond = fargs.remove(0).cast_to(Type::bool());
+        //TODO: actually use the error message passed by the user.
+        let msg = "EXPECTED FAIL";
+        let loc = tcx.codegen_span_option2(span);
+        Stmt::block(
+            vec![
+                Stmt::assert(cond, msg, loc.clone()),
```

Comment:
I do not understand why we are unable to use the error message here. You mention CBMC is not capable of taking non-constant messages, but the one included below is constant, right?

Reply:
The value is a constant. But Rust puts it inside a variable, and then passes the address of the variable. So we need to figure out how to reverse that.

Reply:
https://github.com/model-checking/rmc/issues/236